frontend/src/api/useAccessReview.ts

```diff
@@ -0,0 +1,73 @@
+import { k8sCreateResource } from '@openshift/dynamic-plugin-sdk-utils';
+import * as React from 'react';
+import { ProjectModel, SelfSubjectAccessReviewModel } from '~/api/models';
+import { AccessReviewResourceAttributes, SelfSubjectAccessReviewKind } from '~/k8sTypes';
+import { K8sVerb } from '~/k8sTypes';
+
+const checkAccess = (
+  group: string,
+  resource: string,
+  subresource: string,
+  verb: K8sVerb,
+  name: string,
+  namespace: string,
+): Promise<SelfSubjectAccessReviewKind> => {
```

Comment:
Just inherit the types -- OpenShift Console was written by non-TS engineers in the early days, they learned as they progressed -- types don't need to duplicated.

```suggestion
const checkAccess = ({
  group,
  resource,
  subresource,
  verb,
  name,
  namespace,
}: Required<AccessReviewResourceAttributes>): Promise<SelfSubjectAccessReviewKind> => {
```